src/lib.rs

```diff
@@ -420,21 +420,27 @@ where
 
         // Get span name
         if self.spans {
-            let span = ctx.scope().fold(String::new(), |mut spans, span| {
-                // Add span fields to the base object
-                if let Some(span_object) = span.extensions().get::<Map<String, Value>>() {
-                    object.extend(span_object.clone());
-                }
-                if spans != String::new() {
-                    spans = format!("{}:{}", spans, span.name());
-                } else {
-                    spans = span.name().to_string();
-                }
-
-                spans
+            let span = ctx.current_span().id().and_then(|id| {
```

Comment:
I did a little jiggle around before `0.6`. Do you mind rebasing against lastest `master`, this should work

```rust
            let span = ctx.current_span().id().and_then(|id| {
                ctx.span_scope(id).map(|scope| {
                    scope.from_root().fold(String::new(), |mut spans, span| {
                        // Add span fields to the base object
                        if let Some(span_object) = span.extensions().get::<HashMap<Cow<'static, str>, Value>>() {
                            object.extend(span_object.clone());
                        }
                        if spans != String::new() {
                            spans = format!("{}:{}", spans, span.name());
                        } else {
                            spans = span.name().to_string();
                        }

                        spans
                    })
                })
            });

            if let Some(span) = span {
                object.insert("_span".into(), span.into());
            }
```